shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -894,6 +890,25 @@ protected FlutterEngine getFlutterEngine() {
     return delegate.getFlutterEngine();
   }
 
+  private Bundle cachedMetaData;
```

Comment:
sorry, I don't love this. Don't do any of this and just use https://javadoc.io/static/org.mockito/mockito-core/3.2.4/org/mockito/Mockito.html#spy-T-